config/swagger.yaml

```diff
@@ -339,10 +335,8 @@ paths:
             type: object
             properties:
               count:
-                type: integer
-                minimum: 1
                 description: 'Count'
-                format: int64
+                $ref: '#/definitions/UInt32'
```

Comment:
A path item object is defined [here](https://github.com/OAI/OpenAPI-Specification/blob/8a387ac71f7f31aaff5b910f1e0e99ad12ff3ef0/versions/2.0.md#pathItemObject).

For get operation, the responses object is defined [here](https://github.com/OAI/OpenAPI-Specification/blob/8a387ac71f7f31aaff5b910f1e0e99ad12ff3ef0/versions/2.0.md#responsesObject).

For a specific response code, the value can be `Response Object | Reference Object`. For this 200, it is response object.

The response object defines value of field schema as [schema object](https://github.com/OAI/OpenAPI-Specification/blob/8a387ac71f7f31aaff5b910f1e0e99ad12ff3ef0/versions/2.0.md#schemaObject).

Schema object defines value `properties`. The value of each property is a schema object.

The schema object (namely - the value of object field `count`) can have fields `$ref`, `description`, ... /me thinking of semantics of `$ref`, and how it interacts of other fields 🤔

Reply:
If you have a `$ref` field all other fields are ignored if I remember correctly... So the `description` is really just for documentation...

Reply:
In [definition of schema object](https://github.com/OAI/OpenAPI-Specification/blob/8a387ac71f7f31aaff5b910f1e0e99ad12ff3ef0/versions/2.0.md#schemaObject) I read:
> The following properties are taken directly from the JSON Schema definition and follow the same specifications:
> * $ref - As a [JSON Reference](https://tools.ietf.org/html/draft-pbryan-zyp-json-ref-03)
> * ...
> * description

In [linked JSON reference](https://tools.ietf.org/html/draft-pbryan-zyp-json-ref-03#section-3) I read:
> A JSON Reference is a JSON object, which contains a member named "$ref", which has a JSON string value.  Example:
>    { "$ref": "http://example.com/example.json#/foo/bar" }
> If a JSON value does not have these characteristics, then it SHOULD NOT be interpreted as a JSON Reference.
> The "$ref" string value contains a URI [RFC3986], which identifies the location of the JSON value being referenced.  It is an error condition if the string value does not conform to URI syntax rules. Any members other than "$ref" in a JSON Reference object SHALL be ignored.

So, according to the linked JSON referce a `"$ref"` property makes any other members in the same object were it is to-be-ignored.

In the case, the member to-be-ignored is a description. In cases where more structural members are to-be-ignored, I would not be surprised if there were surprises depending on the tool.

Reply:
Also - parsed by swagger JSON builder, as seen [here](https://github.com/aeternity/aeternity/blob/2f1890932cbd3b9225f46a0fe7846875adebc7c4/apps/aehttp/priv/swagger.json#L410)